### PR TITLE
Kafka2 [DONT MERGE YET]

### PR DESCRIPTION
**Pull request by mikebell90**
This has been labeled 3.0.0 because it supports ONLY Kafka 2.0.0, and once added to otj-parent will be a point of no return. 
* Spring Boot 2.1 uses Kafka 2.0.0 so we'll have to switch then regardless (probably)

Once this is merged and released, you'll be able to test your app against Kafka 2, using the following
* Upgrade to POM 171 (strictly speaking not 100% necessary, but recommended so we have a baseline)
* Add a dependency management entry, forcing otj-kafka 3.0.0
* Add the following to your POM properties
<dep.kafka.version>2.0.0</dep.kafka.version>
* Check to see indeed that your dependency tree is pulling pure kafka 2.0.0
* You may need to also add the dependency management in the otj-kafka POM that adds the zookeeper version change. Let me know if this was needed.

Major changes in Kafka 2.0.0
* All over the place but
* Kafka Streams - supposedly a lot of jiggering here
* All the poll timeouts etc are now deprecated in favor using Duration (yay)
* Significant refactoring and improvements in the (now official Java) Admin Client

**I need folks to test this once it's approved and released. LMK if you know others that should be on the review list**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -19,21 +19,52 @@
...
   <scm>
     <connection>scm:git:git://github.com/opentable/otj-kafka.git</connection>
     <developerConnection>scm:git:git@github.com:opentable/otj-kafka.git</developerConnection>
     <url>http://github.com/opentable/otj-kafka</url>
     <tag>HEAD</tag>
   </scm>
 
+  <properties>
+    <!-- Remove once otj-parent is updated with this version of otj-metrics, and kafka 2.0.0 -->
+    <dep.kafka.version>2.0.0</dep.kafka.version>
```
Comment (mikebell90) [inline pom.xml]: Core change

File: pom.xml [hunk 2]
```diff
@@ -19,21 +19,52 @@
...
   <groupId>com.opentable.components</groupId>
   <artifactId>otj-kafka</artifactId>
   <version>2.8.4-SNAPSHOT</version>
   <description>Kafka integrations component</description>
 
+  <dependencyManagement>
+    <dependencies>
+      <!-- Remove as well -->
+      <dependency>
+        <groupId>org.apache.zookeeper</groupId>
```
Comment (mikebell90) [inline pom.xml]: This was needed - I suspect you guys will need too.

File: src/main/java/com/opentable/kafka/util/OffsetMonitor.java [hunk 3]
```diff
@@ -105,16 +104,26 @@ public void close() {
...
-                .entrySet()
-                .stream()
-                .filter(entry -> entry.getKey().topic().equals(topic))
-                .collect(
-                        Collectors.toMap(
-                                entry -> entry.getKey().partition(),
-                                entry -> (Long) entry.getValue()
-                        )
-                );
+
```
Comment (mikebell90) [inline src/main/java/com/opentable/kafka/util/OffsetMonitor.java]: Admin client changed a bit

File: src/test/java/com/opentable/kafka/KafkaBrokerRuleTest.java [hunk 4]
```diff
@@ -47,7 +48,7 @@ public void testKafkaRule() throws Exception {
 
         try (KafkaConsumer<String, String> consumer = ekb.createConsumer("test")) {
             consumer.subscribe(Collections.singletonList(TEST_TOPIC));
-            ConsumerRecords<String, String> records = consumer.poll(5000);
+            ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(5));
```
Comment (mikebell90) [inline src/test/java/com/opentable/kafka/KafkaBrokerRuleTest.java]: Note how the old poll(long timeout) works but gets .a deprecation warning

**Comment by mikebell90** [comment]
@SegFaultAX See anything scary?

**Comment by SegFaultAX** [comment]
Nothing scarier than backwards incompatibility with established infrastructure. :)

**Pull request merged**